BE/db/index.js

```diff
@@ -0,0 +1,748 @@
+      IN (${ valuesStringSelect });
+    `, tagList);
+
+    return rows;
+  } catch (error) {
+    throw error;
+  }
+}
+
+// CREATE ALL RECIPE_TAGS NECESSARY
+async function addTagsToRecipe(recipeId, tagList) {
+  if (!tagList) {
+    return await getRecipeById(recipeId);
+  }
+
+  try {
+    const createRecipeTagPromises = tagList.map(
+      tag => createRecipeTag(recipeId, tag.id)
+    );
+
+    await Promise.all(createRecipeTagPromises);
+
+    return await getRecipeById(recipeId);
+  } catch (error) {
+    throw error;
+  }
+}
+
+/**
+ * RECIPE_TAGS Methods
+ */
+
+// CREATE RECIPE_TAG IN JUNCTION TABLE FROM DB
+async function createRecipeTag(recipeId, tagId) {
+  try {
+    await client.query(`
+      INSERT INTO recipe_tags(recipeId, tagId)
+      VALUES ($1, $2)
+      ON CONFLICT (recipeId, tagId) DO NOTHING;
+    `, [ recipeId, tagId ]);
+  } catch (error) {
+    throw error;
+  }
+}
+
+/**
+ * REVIEWS Methods
+ */
+
+//GET REVIEW INFO BY ID IN DB
+async function getReviewInfoById(reviewId){
+  try{
+    const {rows: [review]} = await client.query(`
+    SELECT *
+    FROM reviews
+    WHERE id = ${reviewId};
+    `);
+  
+    if (!review) {
+        throw {
+          name: "ReviewNotFoundError",
+          message: "Could not find a review with that reviewId"
+        };
+      }
+    return review;
+  } catch (error){
+    throw (error);
+  }
+}
+
+// GET REVIEW BY ID IN DB
+async function getReviewById(reviewId){
+    try{
+        const review = await getReviewInfoById(reviewId);
+
+        const comments = await getCommentsByReview(reviewId);
+
+        const { rows: [user] } = await client.query(`
+        SELECT id, email, username, name, imgUrl, admin
+        FROM users
+        WHERE id=$1;
+      `, [review.userid]);
+
+        const reviewObject = {
+          ...review,
+          comments: comments,
+          user: user
+        }
+
+        return reviewObject;
+    } catch (error){
+        throw error;
+    }
+}
+
+// GET REVIEWS BY USER IN DB
+async function getReviewsByUser(userId) {
+    try {
+      const { rows: reviewIds } = await client.query(`
+        SELECT id 
+        FROM reviews
+        WHERE userId=${ userId };
+      `);
+  
+      const reviews = await Promise.all(reviewIds.map(
+        review => getReviewById( review.id )
+      ));
+  
+      return reviews;
+    } catch (error) {
+      throw error;
+    }
+  }
+
+// GET REVIEWS BY RECIPE IN DB
+async function getReviewsByRecipe(recipeId) {
+    try {
+      const { rows: reviewIds } = await client.query(`
+        SELECT id 
+        FROM reviews
+        WHERE userId=${ recipeId };
+      `);
+  
+      const reviews = await Promise.all(reviewIds.map(
+        review => getReviewById( review.id )
+      ));
+  
+      return reviews;
+    } catch (error) {
+      throw error;
+    }
+  }
+
+// GET ALL REVIEWS IN DB
+async function getAllReviews() {
+  try {
+    const { rows: reviewIds } = await client.query(`
+      SELECT id
```

Comment:
Why are we not just selecting All from this table for this?

Reply:
Because i wanted the child comments and user info, too, at least for the test function, so it's easier to repurpose the getById instead of rebuilding the return data from scratch